lib/state/terms_and_conditions.dart

```diff
@@ -49,12 +38,13 @@ class TermsAndConditionsAcceptanceState {
 /// State component of the currently accepted and valid Terms & Conditions.
 class TermsAndConditionAcceptance extends Cubit<TermsAndConditionsAcceptanceState> {
   /// Service used to persist the accepted T&C version.
-  final SharedPreferencesService _prefs;
+  final StorageProvider _storage;
```

Comment:
If we go by the naming convention used by block, then I think you should instead create a `TermsAndConditionsRepository` that wraps the `StorageProvider` to expose the terms and conditions specific functionality. Otherwise you expose this class to a lot more functionality (everything storage related) than it actually needs. And then the `StorageProvider` will just expose all the boxes for other classes to use.

Reply:
Worth trying out but beware of overengineering :)

Reply:
I did this refactoring and it came out quite nice. Kudos to you @orhoj 

It also solves one of the comments below. We have a single storage provider with the responsibility of opening connections and expose “boxes”.

The repository wrappers handle the queries and commands. In this case a `TermsAndConditionsRepository` wraps the read, write and update access to the `AcceptedTermsAndConditions`box.
These repository classes could live in network services and have relevant network filtering when accessing data. Since Terms and Condition are global it will live outside these.

Finally the `TermsAndConditionAcceptance` cubit holds the business logic.

I think it came out quite nice and resolved other comments given.